### ISSUE TITLE
[Add Book] possible to add book with negative year

### ISSUE THREAD

**Issue by oscarngncc** [dazzling-sky/comp4111-project#24]
#### Issue
It's possible to add a book with negative value for year:
![image](https://user-images.githubusercontent.com/29769020/83397283-f3dd3480-a42f-11ea-83b3-ca85df3b78bf.png)

Expected output: 400 Bad request or any custom behavior
Actual output: 201 Created with the book actually added to the table

Result added to the database:
![image](https://user-images.githubusercontent.com/29769020/83397311-022b5080-a430-11ea-90e6-a5dbe401a1f8.png)

**Comment by comp4111ta** [comment]
Indeed, that's something that can be improved, yet it is still an undefined behavior. We regard all undefined behaviors as valid in the grading.
